pkg/controller/swiftproxy/swiftproxy_controller.go

```diff
@@ -152,7 +152,11 @@ func (r *ReconcileSwiftProxy) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, fmt.Errorf("failed to list swift proxy pods: %v", err)
 	}
 
-	if err := r.ensureCertificatesExist(swiftProxy, swiftProxyPods, swiftProxy.Status.ClusterIP); err != nil {
+	var alternativeIP []string
+	if swiftProxy.Status.LoadBalancerIP != "" {
+		alternativeIP = append(alternativeIP, swiftProxy.Status.LoadBalancerIP)
+	}
+	if err := r.ensureCertificatesExist(swiftProxy, swiftProxyPods, swiftProxy.Status.ClusterIP, alternativeIP...); err != nil {
```

Comment:
LoadBalancerIP may be assigned with some delay or even may not be assigned at all. 
In case of significant delay I see potential problem.
1) LoadBalancerIP is empty
2) Certificate is created
3) SwiftProxy is started and reads certificate without LoadBalancerIP
4) LoadBalancerIP is set
5) Certificate is updated
6) SwiftProxy may be still using certificate without LoadBalancerIP. I'm not sure if it can reload those files automatically.

Could you please check that?
If that is the case swiftproxy controller needs to wait until LoadBalancerIP is set or service needs to be reloaded